File: Day70/Part6/turn_priority.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        long long total_turns = 0;
        for (int i = 0; i < n; ++i) {
            int p;
            cin >> p;
            while (p > 0) {
                p /= 2;
                total_turns++;
            }
        }
        cout << total_turns << endl;
    }
    return 0;
}

File: Day70/Part6/turn_priority.py
def solve():
    try:
        line1 = input().split()
        if not line1:
            return
        n = int(line1[0])
        
        line2 = input().split()
        priorities = [int(x) for x in line2]
        
        total_turns = 0
        for p in priorities:
            while p > 0:
                p //= 2
                total_turns += 1
                
        print(total_turns)
            
    except EOFError:
        pass

if __name__ == "__main__":
    solve()
